extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlApply.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Alipay.com Inc.
+ * Copyright (c) 2004-2020 All Rights Reserved.
+ */
+package io.kubernetes.client.extended.kubectl;
+
+import com.flipkart.zjsonpatch.JsonDiff;
+import com.google.common.base.Strings;
+import io.kubernetes.client.common.KubernetesListObject;
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.custom.V1Patch;
+import io.kubernetes.client.extended.kubectl.exception.KubectlException;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.util.PatchUtils;
+import io.kubernetes.client.util.Yaml;
+import io.kubernetes.client.util.generic.GenericKubernetesApi;
+import io.kubernetes.client.util.generic.KubernetesApiResponse;
+import io.kubernetes.client.util.generic.options.PatchOptions;
+
+import java.io.IOException;
+
+public class KubectlApply<ApiType extends KubernetesObject> implements Kubectl.Executable<ApiType> {
+
+    private final ApiClient apiClient;
+    private final Class<ApiType> apiTypeClass;
+    private Mode applyMode;
+
+    private String apiGroup;
+    private String apiVersion;
+    private String resourceNamePlural;
+
+    private String yamlContent;
+    private String fieldManager;
+    private boolean forceConflict;
+
+
+    KubectlApply(Class<ApiType> apiTypeClass, ApiClient apiClient) {
+        this.apiClient = apiClient;
+        this.apiTypeClass = apiTypeClass;
+
+        this.applyMode = Mode.SERVER_SIDE_APPLY;
+        this.forceConflict = false;
+    }
+
+
+    public KubectlApply<ApiType> apiGroup(String apiGroup) {
+        this.apiGroup = apiGroup;
+        return this;
+    }
+
+    public KubectlApply<ApiType> apiVersion(String apiVersion) {
+        this.apiVersion = apiVersion;
+        return this;
+    }
+
+    public KubectlApply<ApiType> resourceNamePlural(String resourceNamePlural) {
+        this.resourceNamePlural = resourceNamePlural;
+        return this;
+    }
+
+
+    public KubectlApply<ApiType> mode(Mode applyMode) {
+        // NOTE: currently only server-side-apply supported
+        this.applyMode = applyMode;
+        return this;
+    }
+
+    public KubectlApply<ApiType> fieldManager(String fieldManager) {
+        this.fieldManager = fieldManager;
+        return this;
+    }
+
+    public KubectlApply<ApiType> forceConflict() {
+        this.forceConflict = true;
+        return this;
+    }
+
+    public KubectlApply<ApiType> load(String yamlContent) {
```

Comment:
Should we take a version that takes an `InputStream` also?

Reply:
can we currently stay align w/ `KubectlCreate`? then we can add a `resource(InputStream s)` in a follow-up